File: Búsqueda/Secuencial/secuencial.swift
import Foundation

func busquedaSecuencial(_ arreglo: [Int], _ objetivo: Int) -> Int? {
    for (indice, elemento) in arreglo.enumerated() {
        if elemento == objetivo {
            return indice
        }
    }
    return nil
}

// Ejemplo de uso
let numeros = [4, 2, 7, 1, 9, 3]
if let indice = busquedaSecuencial(numeros, 7) {
    print("Elemento encontrado en el índice \(indice)")
} else {
    print("Elemento no encontrado")
}

File: Búsqueda/Secuencial/secuencial.rs
fn busqueda_secuencial(arr: &[i32], objetivo: i32) -> Option<usize> {
    for (indice, &elemento) in arr.iter().enumerate() {
        if elemento == objetivo {
            return Some(indice);
        }
    }
    None
}

fn main() {
    let arreglo = [1, 2, 3, 4, 5];
    let objetivo = 3;

    match busqueda_secuencial(&arreglo, objetivo) {
        Some(indice) => println!("Elemento encontrado en el índice: {}", indice),
        None => println!("Elemento no encontrado en el arreglo"),
    }
}